### PR TITLE
Whcm   buttons

### LINKED ISSUE #2529: Support for Windows High Contrast Themes
# 🙋 Feature Request

We need to Support Windows High Contrast Mode in our web PDF viewer. Currently UI elements don't adapt Windows High Contrast Mode, which is a blocker in completing support for High Contrast Mode. 
As most of our UI elements use react-spectrum CSS classes for styling, Please help to add support for High Contrast in react-spectrum CSS classes.

## 🤔 Expected Behavior
UI elements should adapt as per Windows High Contrast Mode.

There are 8 colors available in a High Contrast theme. So, UI elements should adapt colors according to the theme 

## 😯 Current Behavior

UI elements don't adapt as per High Contrast themes

## 💁 Possible Solution

Add support for high contrast mode in existing CSS classes of react-spectrum

## 🔦 Context

We need to Support Windows High Contrast Mode in our web PDF viewer. Currently UI elements don't adapt Windows High Contrast Mode, which is a blocker in completing support for High Contrast Mode. 

## 💻 Examples

Design mocks for High Contrast mode are already present in React-spectrum documentation.

## 🧢 Your Company/Team

Adobe/DocumentCloud

### PR DESCRIPTION
**Pull request by jnurthen**
Partially resolves #2529 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

* Enable WHCM 
* Run button stories
* Compare to XD designs
  * https://spectrum.adobe.com/static/Windows-High-Contrast-Kits/Action-button-WindowsHighContrast.xd
  * https://spectrum.adobe.com/static/Windows-High-Contrast-Kits/Button-WindowsHighContrast.xd

### Button Before 
Screen Shot 2021-12-01 at 4 48 18 PM
### Button After
Screen Shot 2021-12-01 at 4 46 47 PM
### ToggleButton Before
Screen Shot 2021-12-01 at 4 48 40 PM
### ToggleButton After
Screen Shot 2021-12-01 at 4 47 30 PM
### LogicButton Before
Screen Shot 2021-12-01 at 4 49 04 PM
### LogicButton After
Screen Shot 2021-12-01 at 4 49 29 PM

### CODE REVIEW

**Comment by LFDanLu** [comment]
@jnurthen my XD is broken even after a reinstall, I'll try and debug it soon. In the meantime can you confirm that a unfocused CTA button should have the same color fill as the left color picker box for "Button Text" in the Windows High Contrast setting? The screenshot you attached shows the CTA button while focused.

This is mine
CTA WHCM

File: packages/@adobe/spectrum-css-temp/components/button/skin.css [hunk 1]
```diff
@@ -1019,3 +1019,381 @@ governing permissions and limitations under the License.
...
+      color: ButtonText;
+    }
+    &.is-active {
+      color: ButtonText;
+    }
+  }
+  .spectrum-ActionButton {
+    &:focus-ring,
+    &.is-focused  {
+        outline: CanvasText solid 3px;
```
Comment (LFDanLu) [inline packages/@adobe/spectrum-css-temp/components/button/skin.css:1353]: The focus ring for action buttons seems a bit thick compared to the XD files
Local:
png2

XD:
image

Comment (LFDanLu) [reply packages/@adobe/spectrum-css-temp/components/button/skin.css:1353]: Note: Made a change to remove this here: https://github.com/adobe/react-spectrum/commit/b3d5e334dfa5610bb00c5d0917fa73f461c4e825, will need to ask Spectrum if it should be a halo focus ring or not

From XD (emphasized and selected):
image

From local:
fafawfawf

**Review by snowystinger — APPROVED** [review]
lgtm, worked with @LFDanLu I can't currently test on a windows machine, but this is what I expect

**Review by devongovett — APPROVED** [review]

**Pull request merged**